src/test/cls_rbd/test_cls_rbd.cc

```diff
@@ -3141,11 +3141,8 @@ TEST_F(TestClsRbd, sparsify)
   ASSERT_EQ(0, sparsify(&ioctx, oid, 16, false));
   std::map<uint64_t, uint64_t> m;
   bufferlist outbl;
-  std::map<uint64_t, uint64_t> expected_m = {{0, 0}};
+  std::map<uint64_t, uint64_t> expected_m = {};
   bufferlist expected_outbl;
-  if (sparse_read_supported) {
-    expected_m = {};
-  }
   ASSERT_EQ((int)expected_m.size(),
             ioctx.sparse_read(oid, m, outbl, inbl.length(), 0));
   ASSERT_EQ(m, expected_m);
```

Comment:
.... you could change this to something like `ASSERT_TRUE(m == {} || m == {{0, 0}})` and backport to Nautilus

Reply:
@dillaman adapted and repushed.